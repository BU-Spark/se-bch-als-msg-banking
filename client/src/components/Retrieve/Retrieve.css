
/* root table width */
#root {
    --width-table: 700px;
}

.unprocessed-audio {
    /* 1x2 table*/
    display: table;
    /* center */
    margin: 0 auto;
    width: 100%;
    border: none;
}


.th {
  border: 1px solid black;
  text-align: center;
}
.table:hover { background-color: white; }

.table{
    pointer-events: none;
}
.table-first-col{
    width: 70%;
    overflow: hidden;
}

.processed-audio {
    /* 1X2 table*/
    display: table;
    /* center */
    margin: 0 auto;
    width: 90%;
}

.button {
    background-color: #436397;
    border: none;
    color: white;
    padding: 7px;
    text-align: match-parent;
    text-decoration: none;
    display: display-outside;
    font-size: 16px;
    margin: 4px 2px;
    border-radius: 10%;
  }
.collapsible .content {
    padding: 1px;
    background-color: white;
}
.collapsible .header {
    background-color: white;
    padding: 1px;
    width: 100%;
}
