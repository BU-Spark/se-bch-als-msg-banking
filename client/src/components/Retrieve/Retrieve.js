import React, { useEffect, useState } from "react";
import { useAuthState } from "react-firebase-hooks/auth";
import { useHistory } from "react-router";
import { auth, db } from "../firebase";
import axios from "axios";
const url=require('../settings')

function Retrieve() {
  const [user, loading] = useAuthState(auth);
  const [audio, setAudio] = useState([]);
  const history = useHistory();
  const fetchUserAudio = async () => {
    try {
      const query = await db
        .collection("users")
        .where("uid", "==", user?.uid)
        .get();
      const data = await query.docs[0].data();
      if ("audio" in data) {
        setAudio(audio.concat(data.audio));
      } else {
        setAudio([]);
      }
    } catch (err) {
      console.error(err);
      alert("An error occured while fetching user data");
    }
  };
  useEffect(() => {
    if (loading) return;
    if (!user) return history.replace(process.env.PUBLIC_URL + "/");
    fetchUserAudio();
  }, [loading]);
  const downloadClip = async (name) => {
    try {
      const response = await axios.post(
        (url+"/retrieve_audio"),
        {
          fileName: name,
        }
      );
      window.open(response.data);
    } catch (err) {
      console.log(err);
    }
  };
  const deleteClip = async (fireStoreFilename, cloudStorageFileName) => {
    try {
      console.log('called deleteClip...')
      const token = await auth.currentUser.getIdToken();
      const response = await axios.delete(
        (url+"/delete_audio"),
        {
          'data': {
            'fireStoreFilename': fireStoreFilename,
            'cloudStorageFileName': cloudStorageFileName,
            'Authorization': token
          }
        }
      );
      console.log('Response from delete audio:');
      console.log(response)
    } catch (err) {
      console.log(err);
    }
  };
  var audioList = audio.map((value) => {
    const res = Object.entries(value);
    console.log(`res: ${res}`);
    console.log(res);
    return (
      <div>
        <button
          className="align-center mt-2"
          key="{res}"
          onClick={() => downloadClip(res[0][1])}
        >
          {res[0][0]}
        </button>
        <br />
      </div>
    );
  });
  return (
    <>
      <h1 className="dashboard-header text-center">Rediscover Your Voice</h1>
      <h2 className="dashboard-header text-center">Click to Download!</h2>
      <br />
      <button onClick={() => deleteClip('testEditedchunk-0.WAV', 'Audiodbdd10e0-6088-11ed-b642-df8bb8090224.wav')}>test button</button>
      <div className="col-md-12 text-center">{audioList}</div>
    </>
  );
}

export default Retrieve;
