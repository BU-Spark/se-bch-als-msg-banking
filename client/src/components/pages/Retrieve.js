import React, { useEffect, useState } from "react";
import { useAuthState } from "react-firebase-hooks/auth";
import { useHistory } from "react-router";
import { auth, db, logout } from "../firebase";

import {Button} from 'react-bootstrap';

function Retrieve() {
    const [user, loading, error] = useAuthState(auth);
    const [audio, setAudio] = useState([]);
    const history = useHistory();
    const fetchUserAudio = async () => {
        try {
          const query = await db
            .collection("users")
            .where("uid", "==", user?.uid)
            .get();
          const data = await query.docs[0].data();
          if ("audio" in data){
            setAudio( data.audio);
          }
          else{
            setAudio(audio => [])
          }
        } catch (err) {
          console.error(err);
          alert("An error occured while fetching user data");
        }
      };
      useEffect(() => {
        if (loading) return;
        if (!user) return history.replace(process.env.PUBLIC_URL +"/");
        fetchUserAudio();
      },[loading]);
      console.log(audio[0])
     if (audio.length >0){
      var audioList = audio.map(clip =>{
            return <li key="{clip}">{clip.key}</li>;
          })
    }
    return (
        <>
        <h1 className="dashboard-header text-center">
            Rediscover your voice
        </h1>
        <h2 className="dashboard-header text-center">
            Click to download your voice!
        </h2>
        <br/>
        {audio ? audio.map(clip =>
             <li key="{clip}">{clip}</li>
          ): 'Loading...'}
        {/* {audio.length > 0 &&
        <ul>{audioList}</ul> */}
{/* } */}
        </>
    )

}

export default Retrieve;