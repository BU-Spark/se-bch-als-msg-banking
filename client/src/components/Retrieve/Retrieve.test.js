//THis will be test for Retrieve.js
// Imports
import React from "react";
import Retrieve from "./Retrieve";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import 'regenerator-runtime/runtime'

// Tests:
// 1. Test that the Retrieve component renders

configure({adapter: new Adapter()});
test("Retrieve", () => {
    const wrapper = shallow(<Retrieve />);
    expect(wrapper).toMatchSnapshot();
}
);
