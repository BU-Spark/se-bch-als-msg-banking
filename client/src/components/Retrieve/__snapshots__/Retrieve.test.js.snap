// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Retrieve 1`] = `ShallowWrapper {}`;
